### PR TITLE
Refactor/admin single user view

### PR DESCRIPTION
**Pull request by CodeNamedRobin**
# Description
Refactors the single user view for admins. Refactors the user edit form to use the new form implementation, and clears up a lot of functional code in the main component, moving it to separate components. Below is what the page looks like now.
image

## Related issues/external references
One of many #273 
One of many #228 (admin single user view having functional code)
One of many #228 (removing changeinfocomponent)
## Types of changes

- New feature _(non-breaking change which adds functionality)_
- Style _(Change that do not affect the functionality of the code)_

### CODE REVIEW

**Comment by JustSamuel** [comment]
Please add screenshots highlight your changes :(

**Comment by SuperVK** [comment]
image
Nickname should be able to be null right?

Some translations missing as well. (Header, confirm message)

(bit irrelevant for here cause the backend allows it) Are we allowed to change the names of members account from GEWIS?

File: apps/dashboard/src/modules/seller/components/ProductActionExistingProductForm.vue
Comment (SuperVK) [inline apps/dashboard/src/modules/seller/components/ProductActionExistingProductForm.vue:1]: Why?

File: apps/dashboard/src/modules/seller/components/ProductActionImageForm.vue
Comment (SuperVK) [inline apps/dashboard/src/modules/seller/components/ProductActionImageForm.vue:1]: why?

**Comment by SuperVK** [comment]
Nickname cannot be null is still an issue, to replicate:
1. click on edit
2. click on username
3. de-focus the username field
4. voila

File: apps/dashboard/src/modules/admin/components/users/AdminUserInfoCard.vue [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    firstName: p.firstName,
+    lastName: p.lastName,
+    email: p.email,
+    nickname: p.nickname,
+    isActive: p.active,
+    userType: userTypes.value.find(ut => ut.name === p.type)?.value || undefined,
+    ofAge: p.ofAge,
+    canGoIntoDebt: p.canGoIntoDebt,
+  };
+  form.context.resetForm({ values });
```
Comment (JustSamuel) [inline apps/dashboard/src/modules/admin/components/users/AdminUserInfoCard.vue]: prefer `{ ....p,` and only specify things you change, so in this case
```ts
{
  ...p,
  userType: userTypes.value.find(ut => ut.name === p.type)?.value || undefined,
  isActive: p.active,
}
```

File: apps/dashboard/src/modules/admin/components/users/forms/UserEditForm.vue [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+               :disabled="!edit"
+               id="name" placeholder="Doe" type="text" />
+    <InputSpan :label="$t('userDetails.Nickname')"
+               :value="form.model.nickname?.value.value"
+               :attributes="form.model.nickname?.attr.value"
+               @update:value="form.context.setFieldValue('nickname', $event)"
+               :errors="form.context.errors.value.nickname"
+               :disabled="!edit"
+               id="name" :placeholder="$t('userDetails.Nickname')" type="text" />
+    <!-- Use user.value.type instead of user.type -->
```
Comment (JustSamuel) [inline apps/dashboard/src/modules/admin/components/users/forms/UserEditForm.vue]: ?

**Review by JustSamuel — CHANGES_REQUESTED** [review]

File: apps/dashboard/src/modules/seller/components/ProductActionImageForm.vue
Comment (CodeNamedRobin) [reply apps/dashboard/src/modules/seller/components/ProductActionImageForm.vue:1]: Okay you got me, it was a lint-fix

File: apps/dashboard/src/modules/admin/components/users/forms/UserEditForm.vue (hunk 2 above)
Comment (CodeNamedRobin) [reply apps/dashboard/src/modules/admin/components/users/forms/UserEditForm.vue]: Fixed in latest commit.

File: apps/dashboard/src/modules/admin/components/users/AdminUserInfoCard.vue (hunk 1 above)
Comment (CodeNamedRobin) [reply apps/dashboard/src/modules/admin/components/users/AdminUserInfoCard.vue]: Fixed in latest commit

**Comment by CodeNamedRobin** [comment]
> Nickname cannot be null is still an issue, to replicate:
> 
>     1. click on edit
> 
>     2. click on username
> 
>     3. de-focus the username field
> 
>     4. voila

This has been fixed in the latest commit.

**Review by JustSamuel — APPROVED** [review]

**Pull request merged**